universum/modules/vcs/perforce_vcs.py

```diff
@@ -499,34 +507,34 @@ def map_local_path_to_depot(self, report):
 
     @make_block("Revert workspace to depot state")
     @catch_p4exception()
-    def copy_cl_files_and_revert(self):
+    def copy_cl_files_and_revert(self) -> List[Tuple[Optional[str], Optional[str], Optional[str]]]:
         self.unshelved_files = self.p4.run_opened()
-        unshelved_path = []
+        unshelved_path: List[Tuple[Optional[str], Optional[str], Optional[str]]] = []
 
-        unshelved_filtered = [item for item in self.unshelved_files if item["action"] != "move/delete"]
+        unshelved_filtered: List[Dict[str, str]] =\
+            [item for item in self.unshelved_files if item["action"] != "move/delete"]
 
         for item in unshelved_filtered:
-            if item["action"] != "delete":
+            relative: Optional[str] = None
+            copied: Optional[str] = None
+            absolute: Optional[str]
```

Comment:
this is to avoid re-declaration, which mypy doesn't like

Reply:
I don't understand what you are talking about, but it is fine if it is needed for mypy.
Maybe add a comment?

Reply:
previously relative, copied, and absolute were declared inside branches
now they are declared before branching, updated inside branches, and used after branch
slighlty less code duplication